examples/ZMP/LinearInvertedPendulum.m

```diff
@@ -195,11 +195,33 @@
   end
   
   methods (Static)
+    function [A, B, C, D, Q, R, Q1, R1, N] = setupLinearSystem(h_over_g, Qy)
+      A = [zeros(2),eye(2);zeros(2,4)]; 
+      B = [zeros(2);eye(2)];
+      C = [eye(2), zeros(2)];
+      D = -h_over_g*eye(2);
+
+      % LQR costs for the output system
+      Q = Qy(5:6,5:6); % for consistency with the other solution methods
+      R = zeros(2);
+
+      % Convert to costs in xbar, u
+      Q1 = C'*Q*C;
+      R1 = R + D'*Q*D;
+      N = C'*Q*D;
+    end
+
     function [ct,Vt,comtraj] = ZMPtrackerClosedForm(h,dZMP,options)
       if nargin<3 options = struct(); end
-      if ~isfield(options,'compute_lyapunov') options.compute_lyapunov = (nargout>1); end
-      if ~isfield(options, 'Qy') options.Qy = diag([0,0,0,0,1,1]); end
-      
+      options = applyDefaults(options, struct('compute_lyapunov', (nargout>1),...
+                                              'Qy', diag([0,0,0,0,1,1]),...
+                                              'use_lqr_cache', false,...
+                                              'lqr_cache_com_height_resolution', 0.01,...
+                                              'build_control_objects', true));
+      if options.use_lqr_cache
```

Comment:
wouldn't it be cleaner to just cache here based on h, instead of caching on the linear system (more general but also slower/more complex) but with a method that lives inside the ZMP dir (so it not used generally)